e2e/nomostest/gitproviders/bitbucket.go

```diff
@@ -93,154 +97,155 @@ func (b *BitbucketClient) SyncURL(name string) string {
 
 // CreateRepository calls the POST API to create a remote repository on Bitbucket.
 // The remote repo name is unique with a prefix of the local name.
-func (b *BitbucketClient) CreateRepository(localName string) (string, error) {
-	u, err := uuid.NewRandom()
-	if err != nil {
-		return "", fmt.Errorf("failed to generate a new UUID: %w", err)
-	}
-
-	// strip all hyphens from localName and uuid before appending
-	localName = strings.ReplaceAll(localName, "/", "")
-	localName = strings.ReplaceAll(localName, "-", "")
-	if len(localName) > localNameLength {
-		localName = localName[:localNameLength]
-	}
-
-	uuid := strings.ReplaceAll(u.String(), "-", "")
-
-	repoName := localName + uuid
-	if len(repoName) > repoNameMaxLength {
-		repoName = repoName[:repoNameMaxLength]
-	}
+func (b *BitbucketClient) CreateRepository(name string) (string, error) {
+	fullName := b.fullName(name)
+	repoURL := fmt.Sprintf("https://api.bitbucket.org/2.0/repositories/%s/%s", b.workspace, fullName)
 
 	// Create a remote repository on demand with a random localName.
 	accessToken, err := b.refreshAccessToken()
 	if err != nil {
 		return "", err
 	}
 
-	out, err := exec.Command("curl", "-sX", "POST",
-		"-H", "Content-Type: application/json",
-		"-H", fmt.Sprintf("Authorization:Bearer %s", accessToken),
-		"-d", fmt.Sprintf("{\"scm\": \"git\",\"project\": {\"key\": \"%s\"},\"is_private\": \"true\"}", bitbucketProject),
-		fmt.Sprintf("https://api.bitbucket.org/2.0/repositories/%s/%s", b.workspace, repoName)).CombinedOutput()
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
 
+	req, err := http.NewRequest(http.MethodGet, repoURL, nil)
```

Comment:
minor nit: the http request logic could be broken out to a helper function to improve the flow/readability